web/app.js

```diff
@@ -1545,32 +1545,21 @@ const PDFViewerApplication = {
   async _initializeAutoPrint(pdfDocument, openActionPromise) {
     const [openAction, jsActions] = await Promise.all([
       openActionPromise,
-      !this.pdfViewer.enableScripting ? pdfDocument.getJSActions() : null,
+      this.pdfViewer.enableScripting ? null : pdfDocument.getJSActions(),
     ]);
 
     if (pdfDocument !== this.pdfDocument) {
       return; // The document was closed while the auto print data resolved.
     }
-    let triggerAutoPrint = false;
-
-    if (openAction?.action === "Print") {
-      triggerAutoPrint = true;
-    }
     if (jsActions) {
-      for (const name in jsActions) {
-        if (jsActions[name]) {
-          console.warn("Warning: JavaScript support is not enabled");
-          break;
-        }
-      }
-
-      // Hack to support auto printing.
-      triggerAutoPrint ||= !!(
-        jsActions.OpenAction && AutoPrintRegExp.test(jsActions.OpenAction)
-      );
+      console.warn("Warning: JavaScript support is not enabled");
```

Comment:
When JS is disabled we must iterate on the actions to find one matching the regex.
We should do the same as the js sandbox and exclude some of them:
https://github.com/mozilla/pdf.js/blob/master/src/scripting_api/doc.js#L106-L111
Just to be sure, I created this pdf:
[autoprint_foobar.pdf](https://github.com/mozilla/pdf.js/files/12315436/autoprint_foobar.pdf)
which just contains an action called `foobar`, in Acrobat, the autoprint is triggered.